File: apiEZPZ/src/main/java/com/example/apiezpz/comment/repository/CommentRepository.java
package com.example.apiezpz.comment.repository;

import com.example.apiezpz.comment.domain.Comment;
import jakarta.transaction.Transactional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;

@Repository
public interface CommentRepository extends JpaRepository<Comment, Long> {
    List<Comment> findByPostIdOrderByCreatedAtDesc(Long postId);
    @Transactional
    void deleteByWriter(String writer);
}

File: apiEZPZ/src/main/kotlin/com/example/apiezpz/comment/repository/CommentRepository.kt
package com.example.apiezpz.comment.repository

import com.example.apiezpz.comment.domain.Comment
import jakarta.transaction.Transactional
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CommentRepository : JpaRepository<Comment, Long> {
    fun findByPostIdOrderByCreatedAtDesc(postId: Long): List<Comment>
    
    @Transactional
    fun deleteByWriter(writer: String)
} 